fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/event/AbstractEventCreationFragment.java

```diff
@@ -55,33 +57,80 @@ public abstract class AbstractEventCreationFragment extends Fragment {
   private EditText startTimeEditText;
   private EditText endTimeEditText;
 
+  private boolean toBeCleared = true;
+
   public void setDateAndTimeView(View view) {
+    long currentMillis = System.currentTimeMillis();
+    long suggestedEndMillis = currentMillis + 1000 * 60 * 60; // Adding one hour
+    startDate = new GregorianCalendar();
+    startTime = new GregorianCalendar();
+    Date dCurrent =
+        new java.util.Date(currentMillis); // Get's the date based on the unix time stamp
+    startDate.setTime(dCurrent);
+    startTime.setTime(dCurrent);
+    Date dSuggestedEnd = new Date(suggestedEndMillis);
+    endDate = new GregorianCalendar();
+    endTime = new GregorianCalendar();
+    endDate.setTime(dSuggestedEnd);
+    endTime.setTime(dSuggestedEnd);
+
     startDateEditText = view.findViewById(R.id.start_date_edit_text);
     startDateEditText.setInputType(InputType.TYPE_NULL);
+    startDateEditText.setText(dateFormat.format(dCurrent));
 
     endDateEditText = view.findViewById(R.id.end_date_edit_text);
     endDateEditText.setInputType(InputType.TYPE_NULL);
+    endDateEditText.setText(dateFormat.format(dSuggestedEnd));
 
     startTimeEditText = view.findViewById(R.id.start_time_edit_text);
     startTimeEditText.setInputType(InputType.TYPE_NULL);
+    startTimeEditText.setText(timeFormat.format(dCurrent));
 
     endTimeEditText = view.findViewById(R.id.end_time_edit_text);
     endTimeEditText.setInputType(InputType.TYPE_NULL);
+    endTimeEditText.setText(timeFormat.format(dSuggestedEnd));
 
     // Offset the threshold a little to accept current value
     threshold.add(Calendar.MINUTE, -1);
 
     startDateEditText.setOnClickListener(
-        v -> openPickerDialog(new DatePickerFragment(), DatePickerFragment.TAG, this::onStartDate));
+        v -> {
+          clearDates();
+          openPickerDialog(new DatePickerFragment(), DatePickerFragment.TAG, this::onStartDate);
+        });
 
     endDateEditText.setOnClickListener(
-        v -> openPickerDialog(new DatePickerFragment(), DatePickerFragment.TAG, this::onEndDate));
+        v -> {
+          clearDates();
+          openPickerDialog(new DatePickerFragment(), DatePickerFragment.TAG, this::onEndDate);
+        });
 
     startTimeEditText.setOnClickListener(
-        v -> openPickerDialog(new TimePickerFragment(), TimePickerFragment.TAG, this::onStartTime));
+        v -> {
+          clearDates();
+          openPickerDialog(new TimePickerFragment(), TimePickerFragment.TAG, this::onStartTime);
+        });
 
     endTimeEditText.setOnClickListener(
-        v -> openPickerDialog(new TimePickerFragment(), TimePickerFragment.TAG, this::onEndTime));
+        v -> {
+          clearDates();
+          openPickerDialog(new TimePickerFragment(), TimePickerFragment.TAG, this::onEndTime);
+        });
+  }
+
+  private void clearDates() {
```

Comment:
I added a suggested time and date of current time and a duration of 1 hour. But given all the rules when entering dates it was painful to modify in the right order so when the user clicks on dates or times for the first time, we clear the suggested values